addons/website_slides_survey/models/slide_slide.py

```diff
@@ -65,3 +65,28 @@ def _action_set_viewed(self, target_partner, quiz_attempts_inc=False):
                         'slide_partner_id': new_slide_partner.id
                     }
                 )
+
+    def _get_slide_certification_data(self):
+        certification_data = {}
+        if not all(slide.channel_id.is_member for slide in self):
```

Comment:
That's not very consistent with the batch using.
If you're member of the related channel, you get the actual URL of the slide.
If you're not, you get a test entry.

That should always be the case even if you give this method both types of slides.

In other words, the `if` part should be inside the loop.

Reply:
Done